typescript/sdk/src/metadata/agentConfig.ts

```diff
@@ -108,13 +335,12 @@ export function buildAgentConfigNew(
   multiProvider: MultiProvider,
   addresses: ChainMap<HyperlaneDeploymentArtifacts>,
   startBlocks: ChainMap<number>,
-): ChainMap<ChainMetadataForAgent> {
-  const configs: ChainMap<ChainMetadataForAgent> = {};
+): ChainMap<AgentChainMetadata> {
+  const configs: ChainMap<AgentChainMetadata> = {};
   for (const chain of [...chains].sort()) {
-    const metadata = multiProvider.getChainMetadata(chain);
-    const config: ChainMetadataForAgent = {
+    const metadata: ChainMetadata = multiProvider.getChainMetadata(chain);
```

Comment:
I'm surprised more changes/additions weren't required here. I guess all the new fields are optional. Just confirming we don't want to expand the params of this helper fn to support more options? (Maybe that comes later which is fine with me)

Reply:
This will have to change when we go to actually use it, just needed something to satisfy type checking for the moment.

Reply:
Not sure yet, will explore that when the time comes. Probably best to make this just a generator of the chain data, though I would like to point out that all we really need in the config files will probably be the chains. The rest we tend to override in helm with envs as needed.